ATP Charter (#60)
OR13/diem-draft-charter: atp/charter.md

Authenticated Transfer Protocol (ATP) Working Group
===================================================

> The Authenticated Transfer Protocol (ATP) working group will develop a set of standards-track specifications that enable creation of an interoperable global network for publication of self-certifying data. 

Author: ```suggestion
The Authenticated Transfer Protocol (ATP) working group will develop a set of Standards Track specifications that enable creation of decentralized global networks for publication of self-certifying data.

See RFC 9518 for additional context on why decentralization matters.
```

Reviewer: +1 to "interoperable" -> "decentralized"

Does it make sense to reference the RFC in the intro paragraph vs near the bottom of the charter?

Reviewer 2: Some nits: 

* s/a decentralized global networks/a decentralized global network 
* s/standards-track/Standards Track

Author: I think the plural is intentional here... the goal for this protocol is to support multiple use cases, and multiple account systems, I think we do not expect these to all coalesce into one network.

> The Authenticated Transfer Protocol (ATP) working group will develop a set of standards-track specifications that enable creation of an interoperable global network for publication of self-certifying data. 

Reviewer: ```suggestion
The Authenticated Transfer Protocol (ATP) working group will develop a set of standards-track specifications that enable creation of an interoperable global network for publication of self-certifying data. 
Data is published by accounts in data repositories. An account represents a user or actor's identity, can migrate between hosting locations over time, and is identified by a persistent globally-resolvable account identifier.
```

Reviewer 2: This is a great addition. Thanks @dholms

> The protocol enables synchronization of public data across organizational boundaries, while preserving authenticity of content and attribution. 

Author: ```suggestion
The protocol enables synchronization of public data across organizational boundaries, while preserving authenticity of content and attribution to the accounts that have published the content.
```

Author: @boucadair towards your first comment.

Reviewer: Pulled this recommendation into my suggestion as well: https://github.com/ietf-artarea/charters/pull/60/changes#r2691199416

> The protocol enables synchronization of public data across organizational boundaries, while preserving authenticity of content and attribution. 

Reviewer: ```suggestion
The network comprises independent hosts and other participants that synchronize public data across organizational boundaries, while preserving authenticity of content and attribution to the accounts that published the content.
```

Reviewer 2: I like this one as it defines "network" but also clarify the link with accounts.

The network will give small independent participants global visibility and distribution without prior coordination, and will not depend on the existence of any single privileged entity.

The work will be based on draft-holmgren-at-repository and the pre-standardization implementation and deployment experience of multiple parties in the ATP ecosystem. 

> All existing work must undergo a consensus process, and changes may be necessary, but gratuitous changes will be avoided when possible.

Reviewer: ```suggestion
All existing work must undergo a consensus process. Proposed changes to the existing approach must be weighed against their impact on deployed implementations.
```

> All existing work must undergo a consensus process. Proposed changes to the existing approach must be weighed against their impact on deployed implementations.

Author: ```suggestion
Proposed changes to the existing approach must be weighed against their impact on deployed implementations.
```

Repositories are cryptographically verifiable and publicly published.

Cryptographic verification of repository signatures requires a system for resolving account identifiers to public signing keys. 
Account identifiers are also used to resolve the current hosting location. Account identifiers should be globally resolvable and persist across hosting migrations. 
The working group will specify interface requirements and selection criteria for account identifier systems. 
To ensure broad interoperability in the global network, the working group will create a list indicating which identifier systems are recommended as a baseline, and a process for updating that list over time. 
Backwards compatibility will be maintained for the two account identifier systems supported in the currently deployed network (PLC and did:web).

Deliverables for the working group:

- Standards track document(s) describing a repository data structure, an encoding format for structured data records (CBOR and JSON representations), and an export format for transfer of data repositories

> - Standards track document(s) describing a low-latency streaming synchronization protocol

Reviewer: ```suggestion
- Standards track document(s) describing a low-latency synchronization protocol
```

- Standards track document(s) describing a URI scheme ('at:') for persistent references between data records
- Document(s) describing interface requirements and selection criteria for account identifier resolution systems, and defining a public registry and update process for recommending specific account identifier resolution systems

Key goals for ATP are:

> - The streaming protocol should include notification of data deletion

Reviewer: ```suggestion
- The synchronization protocol should include notification of data deletion
```

> - The streaming protocol should allow for alternative network transports

Reviewer: ```suggestion
- The synchronization protocol should allow for alternative network transports
```

- Scale to global network applications
- Support for redistribution of authenticated data via multiple layers of intermediary parties
- Account migration between network locations (eg, hosting providers) does not break references or graph relationships

The following are out of scope for the working group:

- Transfer of non-public or limited-visibility data
- Any application-specific data schemas
- Any social application semantics

The ATP working group will consult the Crypto Forum Research Group (CFRG) and Security Area groups concerning the use and selection of any cryptographic systems. 

> Privacy and Centralization considerations will be documented (eg, per RFC 6973) and will inform design decisions.

Author: ```suggestion
Privacy and Centralization per RFC 6973 and RFC 9518 will inform design decisions.
```

Authenticated Transfer Protocol (ATP) Working Group
===================================================

> The Authenticated Transfer Protocol (ATP) working group will develop a set of Standards Track specifications that enable creation of decentralized global networks for publication of self-certifying data.

Reviewer 2: For internal consistency: please double check "global networks" use vs the use below such as in "interoperability in **the** global network"

Are these referring to the same thing? if so, some alignment is needed.

Author: I think i got this, let me know if you disagree.

The network will give small independent participants global visibility and distribution without prior coordination, and will not depend on the existence of any single privileged entity.

The work will be based on draft-holmgren-at-repository and the pre-standardization implementation and deployment experience of multiple parties in the ATP ecosystem. 
Proposed changes to the existing approach must be weighed against their impact on deployed implementations.

The working group will focus on a general purpose data repository structure for public user data records, and mechanisms for synchronizing repositories across the network. 
Repositories are cryptographically verifiable and publicly published.

Cryptographic verification of repository signatures requires a system for resolving account identifiers to public signing keys. 
Account identifiers are also used to resolve the current hosting location. Account identifiers should be globally resolvable and persist across hosting migrations. 
The working group will specify interface requirements and selection criteria for account identifier systems. 

> To ensure broad interoperability in the global network, the working group will create a list indicating which identifier systems are recommended as a baseline, and a process for updating that list over time. 

Author: ```suggestion
To ensure broad interoperability, the working group will create a list indicating which identifier systems are recommended as a baseline, and a process for updating that list over time. 
```

Reviewer 2: Works for me.

Backwards compatibility will be maintained for the two account identifier systems supported in the currently deployed network (PLC and did:web).

Deliverables for the working group:

- Standards track document(s) describing a repository data structure, an encoding format for structured data records (CBOR and JSON representations), and an export format for transfer of data repositories
- Standards track document(s) describing a low-latency synchronization protocol
- Standards track document(s) describing a URI scheme ('at:') for persistent references between data records
- Document(s) describing interface requirements and selection criteria for account identifier resolution systems, and defining a public registry and update process for recommending specific account identifier resolution systems

> 

Reviewer 2: ```suggestion
- Document(s) describing operational considerations for deployment and interoperability testing.
```

Reviewer 2: to mirror the proposal made by @dholms in the ballot thread.

> Key goals for ATP are:

Reviewer: ```suggestion

Key goals for ATP are:
```

Need a new line for formatting

- The synchronization protocol should include notification of data deletion

> - The synchronization protocol should allow for alternative network transports

Reviewer 2: ```suggestion
- The synchronization protocol should allow for alternative transport layers
```

Reviewer 2: per the discussion in the list.

- Scale to global network applications
- Support for redistribution of authenticated data via multiple layers of intermediary parties
- Account migration between network locations (eg, hosting providers) does not break references or graph relationships

The following are out of scope for the working group:

- Transfer of non-public or limited-visibility data
- Any application-specific data schemas
- Any social application semantics

> The ATP working group will consult the Crypto Forum Research Group (CFRG) and Security Area groups concerning the use and selection of any cryptographic systems. 

Reviewer 2: ```suggestion
The ATP working group will consult the Crypto Forum Research Group (CFRG) and Security Area groups concerning the use and selection of any cryptographic systems. Likewise, the WG will leverage (and liaise as appropriate) with HAPPY WG for transport selection matters.
```
